pyani_plus/classify.py

```diff
@@ -201,17 +206,201 @@ def get_unique_cliques(
+        linewidth=2,
+        linestyle="--",
+        label="Singleton Genomes",
+    )
+    # Customise ax1: set y-label, grid style, and display legend
+    ax1.set_ylabel("Number of \n Genomes", fontsize=10)
+    ax1.grid(True, linestyle="--", linewidth=0.5, alpha=0.7)  # noqa: FBT003
+    ax1.legend()
+
+    # --- The percentage of all genomes that are found in cliques and as singletons at different %identity (ax2) ---
+    total_genomes = clique_counts + singleton_counts  # Total count per identity bin
+    percent_total_genomes = (total_genomes / num_genomes) * 100  # Convert to percentage
+
+    # Plot percentage of genomes (cliques + singletons)
+    ax2.plot(
+        identity_bins[:-1],
+        percent_total_genomes,
+        color="green",
+        linewidth=2,
+        label="% Genomes",
+    )
+    ax2.fill_between(
+        identity_bins[:-1], percent_total_genomes, color="green", alpha=0.3
+    )
+    # Customise ax2: set y-label, grid style, and display legend
+    ax2.set_ylabel("Percentage of \n All Genomes", fontsize=10)
+    ax2.set_ylim(0, 100)
+    ax2.grid(True, linestyle="--", linewidth=0.5, alpha=0.7)  # noqa: FBT003
+    ax2.legend()
+
+    # --- The lifespan (range of identity) of each clique (ax3) ---
+    for _, row in dataframe.iterrows():
+        x_start, x_end = row[f"min_{score}"], row[f"max_{score}"]
+        y_positions = [genome_positions[g] for g in row["members"]]
+        y_min, y_max = min(y_positions), max(y_positions)
+
+        if len(row["members"]) == 1:
+            # Plot single-member groups as dashed horizontal lines
+            ax3.hlines(
+                y=y_min,
+                xmin=x_start,
+                xmax=x_end,
+                colors="grey",
+                linestyles="dashed",
+                linewidth=1.5,
+            )
+        else:
+            # Plot cliques as rectangles with heatmap coloring
+            color = cmap_hot(norm(x_start))
+            rect = patches.Rectangle(
+                (x_start, y_min - 0.4),
+                x_end - x_start,
+                y_max - y_min + 0.8,
+                linewidth=1,
+                edgecolor="black",
+                facecolor=color,
+                alpha=0.6,
+            )
+            ax3.add_patch(rect)
+
+    ax3.set_xlabel(f"{score}")
+    ax3.set_ylabel("Genomes", fontsize=6)
+    ax3.set_yticks(range(num_genomes))
+    ax3.set_yticklabels(
+        genome_positions.keys(), fontsize=font_size
+    )  # Dynamically adjust font size
+    ax3.yaxis.set_label_position("right")
+    ax3.yaxis.tick_right()
+    ax3.set_xlim(math.floor(dataframe[f"min_{score}"].min() * 100) / 100 - 0.01, 1.0)
+    ax3.set_ylim(-1, num_genomes)
+    ax3.axvline(x=0.95, color="red", linewidth=2, linestyle="--")
+    ax3.grid(True, linestyle="--", linewidth=0.5, alpha=0.7)  # noqa: FBT003
+
+    # Save the figure as a PDF file
+    plt.savefig(
+        outdir / f"{method}_classify_plot.pdf", format="pdf", bbox_inches="tight"
+    )
```

Comment:
Can you import the standard list of graphics formats, and loop over them to make PNG etc as well as PDF?